### DISCUSSION TITLE
Flow Chart of API / .json to the DOM

### DISCUSSION THREAD

**Discussion by BrysonMG** [nss-day-cohort-47/C47-Discussions#29] [Class Notes and Discussion]
I find it helpful to chart out the data to be able to follow what's happening a little easier. Below is a chart I made that shows the process of fetching data from the API/.json file, converting it to HTML, and getting it on the DOM. There are a few notes between the sections of code detailing what's going on; however, I wrote them in a way that makes sense to me rather than trying to explain it to an audience. I apologize if my terminology is off, as that is a weak spot for me. Finally, I'm sorry the image is still a little blurry when you zoom in; Miro's exported images are low resolution unless you are paying them. I hope this helps somebody.

![api flow](https://user-images.githubusercontent.com/59521212/110216151-23174b00-7e73-11eb-90b3-f9ad61df54c9.jpg)

I decided to also include a link to the Miro board itself so you can view it without the blur. I've been having to delete a board every time I need to start a new one, so this link may eventually stop working, but I'll leave it untouched for a while so anyone that wants it can save it in some way.

https://miro.com/app/board/o9J_lQicqcM=/

**Comment by BryanNilsen** [comment]
I LOVE LOVE LOVE that you did this!! This looks great.
Did you find that the actual process of breaking out the code like this helped you understand the flow??

Yeah, it would be nice to zoom in and read the code a bit better. It's tough with the free app. Maybe you can share the link to the document for viewing purposes, but restrict editing to just yourself. (Not sure if that's an option)

**Reply by BrysonMG** [reply]
Yes it did help me understand the flow a bit better, but I find it's the actual writing of the code repeatedly that makes it stick.

But I did include a link to the Miro board itself below the attached image, and it should be view-only. Let me know if it doesn't work properly and I'll try and fix it.
